src/playlist-loader.js

```diff
@@ -421,7 +432,15 @@ const PlaylistLoader = function(srcUrl, hls, withCredentials) {
   /**
    * start loading of the playlist
    */
-  loader.load = () => {
+  loader.load = (isFinalRendition) => {
+    let refreshDelay;
+
+    if (isFinalRendition) {
+      refreshDelay = loader.media() ? loader.media().targetDuration * 1000 : 10 * 1000;
+      window.clearTimeout(mediaUpdateTimeout);
+      mediaUpdateTimeout = window.setTimeout(loader.load(false), refreshDelay);
```

Comment:
`setTimeout` needs to be passed a callback that calls `loader.load(false)`. You could also bind a call to load with false passed, `loader.load.bind(null, false)`